src/Aspire.Hosting/Publishing/PublishingActivityProgressReporter.cs

```diff
@@ -73,13 +73,31 @@ internal PublishingStep(string id, string title)
     /// The progress reporter that created this step.
     /// </summary>
     internal IPublishingActivityProgressReporter? Reporter { get; set; }
+
+    /// <summary>
+    /// Completes the step automatically when disposed if not already completed.
+    /// </summary>
+    public async ValueTask DisposeAsync()
+    {
+        if (Reporter is not null && !IsComplete)
+        {
+            try
+            {
+                await Reporter.CompleteStepAsync(this, Title, isError: false, CancellationToken.None).ConfigureAwait(false);
```

Comment:
We should add IsError to the Step that would make it easier to set the state of the current step.

Also double completing will not throw InvalidOperationException.